### PR TITLE
Test judge pipeline

### LINKED ISSUE #112: Add tests for judges pipeline
## Task Description

- Judges pipeline needs appropriate tests written for functions that don't have external dependencies

## User Stories

- As a developer, I need to test my code, so I know that everything is working as expected.

## Relevant Files [If Available] 

- `pipeline.py`, `test_pipeline.py`

### PR DESCRIPTION
**Pull request by luciegamby-sigmalabs**
Added tests for judge pipeline. I did test things like loading to the database or the actual scraping itself, since this relies on external sources.

Closes #112

### CODE REVIEW

File: judges/test_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+"""
+
+
+@pytest.mark.parametrize("input_date, expected_date", [("12-03-2024", "2024-03-12"),
+                                                       ("12-03-24", "2024-03-12"),
+                                                       ("12-Mar-24", "2024-03-12")])
+def test_convert_date_returns_correct_format(input_date, expected_date):
+    """Tests that the dates are returned in the correct format."""
+
+    assert convert_date(input_date) == expected_date
```
Comment (MahinRahman8901) [inline judges/test_pipeline.py:19]: solid test maybe use more dates just to double check

File: judges/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+def test_extracting_name_gender_returns_correct_types():
+    """Tests return types are correct."""
+
+    extraction = extract_name_gender(
+        "His Honour Judge Fizz", "Honour Judge")
+    assert len(extraction) == 2
+    assert isinstance(extraction[0], str)
+    assert isinstance(extraction[1], str)
+    assert len(extraction[1]) == 1
```
Comment (MahinRahman8901) [inline judges/test_pipeline.py:54]: another good test

File: judges/test_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+"""
+Testing transform_df
+"""
+
+
+def test_transform_df_returns_data_frame():
+    """Tests a DataFrame is returned."""
+
+    judge_data = [{"judge": "His Honour Judge Fizz",
+                  "appointment": "12-03-2024"}]
```
Comment (MahinRahman8901) [inline judges/test_pipeline.py:66]: maybe add more sample data?

**Review by MahinRahman8901 — APPROVED** [review]

File: judges/test_pipeline.py (hunk 3 above)
Comment (luciegamby-sigmalabs) [reply judges/test_pipeline.py:66]: Good catch, I'll parametrize this and add more data that way

**Pull request merged**